lib/dao.js

```diff
@@ -468,6 +468,18 @@ DataAccessObject.updateOrCreate = DataAccessObject.upsert = function upsert(data
   var Model = this;
   var connector = Model.getConnector();
 
+  if (Array.isArray(data)) {
+    var isBulkCreate = data.every(function(item){
```

Comment:
@bajtos, just want to confirm:
I want to remove this entirely now right? In the end, we will simply throw an error upon any array input.

Please let me know if this is correct and I will update the PR.